### PR TITLE
RD-14031-Pure-python-script-instrumentation-bad-experience

### PR DESCRIPTION
**Pull request by eugene-lumigo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,29 +59,50 @@ Replace `<service name> with the desired name of the service`.
...
-**Note:** The instructions in this section are mutually exclusive with those provided in the [No-code activation](#no-code-activation) section.
+**Note:** The instructions in this section are mutually exclusive with those provided in the [No-code activation](#no-code-activation) and [Decorator-based activation](#decorator-based-activation) sections.
 
 Import `lumigo_opentelemetry` at the beginning of your main file:
 
 ```python
 import lumigo_opentelemetry
 ```
 
+### Decorator-based activation
```
Comment (sagivoululumigo) [inline README.md:88]: I wouldn't call the `Decorator based activation`, this is useful for instrumenting simple scripts (or just scripts).
Why not call it `Script instrumentation`?
@lir-lumigo WDYT?

Comment (sagivoululumigo) [inline README.md:88]: BTW this should be a sub-section of the manual instrumentation

File: README.md [hunk 2]
```diff
@@ -59,29 +59,50 @@ Replace `<service name> with the desired name of the service`.
...
+
+For simple Python scripts that are not built around Lumigo or OpenTelemetry-instrumented libraries, you can use the `@lumigo_wrapped` decorator for activation. This approach is especially useful for standalone scripts, cron jobs, or similar use cases.
```
Comment (sagivoululumigo) [inline README.md:90]: You need to first mention that by default if using the autowrapping or manual activation then the script will be instrumented, but there are limitations such as:
* The last spans might be lost if the script ends before they are sent
* If no outbound requests are made (For example if the code crashes before the first call is made) then the script run wont generate any traces in lumigo

File: README.md [hunk 3]
```diff
@@ -212,6 +233,35 @@ from opentelemetry.trace import get_current_span
 get_current_span().add_event('<error-message>', {'lumigo.type': '<error-type>'})
 ```
 
+### Using the `@lumigo_wrapped` Decorator
+
+The `@lumigo_wrapped` decorator is a convenient way to add OpenTelemetry tracing to functions in scripts that are not using libraries or frameworks instrumented by Lumigo or OpenTelemetry.
```
Comment (sagivoululumigo) [inline README.md:238]: I would call it lumigo tracing, no need to talk about the underlying otel

File: README.md [hunk 4]
```diff
@@ -212,6 +233,35 @@ from opentelemetry.trace import get_current_span
...
+
+When applied, the decorator creates an OpenTelemetry span for the decorated function, adding attributes such as:
+
+- `input_args`: The positional arguments passed to the function.
+- `input_kwargs`: The keyword arguments passed to the function.
+- `return_value`: The value returned by the function.
+
+This approach is particularly useful for standalone Python scripts or cron jobs where use of the [OpenTelemetry SDK](https://opentelemetry.io/docs/languages/java/) is not feasible.
```
Comment (sagivoululumigo) [inline README.md:246]: Why are you referencing java here?

**Review by sagivoululumigo — CHANGES_REQUESTED** [review]
See notes + add a unit test for the lumigo wrapped

File: README.md [hunk 5]
```diff
@@ -82,6 +82,30 @@ Import `lumigo_opentelemetry` at the beginning of your main file:
...
+
+See the [Using the `@lumigo_wrapped` Decorator](#using-the-lumigo_wrapped-decorator) section for more details.
+
+#### Limitations
+
+When activating the tracer, consider the following limitations:
+
+1. **Loss of the spans**: The final spans might not be sent to Lumigo if the script ends abruptly or before they can be flushed. To mitigate this, you can use the `force_flush` method on the tracer provider (see [Ensure spans are flushed to Lumigo before shutdown](#ensure-spans-are-flushed-to-lumigo-before-shutdown)).
+
+2. **No traces generated without outbound requests**: If no outbound requests are made (e.g., the code crashes before the first call is executed), the script run will not generate any traces in Lumigo. This limitation is inherent to the OpenTelemetry behavior and the need for spans to be triggered by instrumentation or custom tracing calls.
```
Comment (lir-lumigo) [inline README.md:107]: @eugene-lumigo does this limitation also apply when using the wrapper?

File: README.md [hunk 6]
```diff
@@ -82,6 +82,30 @@ Import `lumigo_opentelemetry` at the beginning of your main file:
 import lumigo_opentelemetry
 ```
 
+##### Script instrumentation
```
Comment (lir-lumigo) [inline README.md:85]:
```suggestion
#### Script instrumentation
```
Trying to reduce the depth of titles here

File: README.md [hunk 7]
```diff
@@ -82,6 +82,30 @@ Import `lumigo_opentelemetry` at the beginning of your main file:
...
+from lumigo_opentelemetry import lumigo_wrapped
+
+@lumigo_wrapped
+def your_function():
+    pass
+```
+
+See the [Using the `@lumigo_wrapped` Decorator](#using-the-lumigo_wrapped-decorator) section for more details.
+
+#### Limitations
```
Comment (lir-lumigo) [inline README.md:101]: Do these limitations apply to all activation methods?

File: README.md [hunk 8]
```diff
@@ -212,6 +236,35 @@ from opentelemetry.trace import get_current_span
...
+
+The `@lumigo_wrapped` decorator is a convenient way to add Lumigo tracing to functions in scripts that are not using libraries or frameworks instrumented by Lumigo or OpenTelemetry.
+
+When applied, the decorator creates an OpenTelemetry span for the decorated function, adding attributes such as:
+
+- `input_args`: The positional arguments passed to the function.
+- `input_kwargs`: The keyword arguments passed to the function.
+- `return_value`: The value returned by the function.
+
+This approach is particularly useful for standalone Python scripts or cron jobs where use of the [OpenTelemetry SDK](https://opentelemetry.io/docs/languages/python/) is not feasible.
```
Comment (lir-lumigo) [inline README.md:249]: @eugene-lumigo I think we can remove this line

**Pull request merged**